utils/cp_utils.py

```diff
@@ -60,14 +60,17 @@ def run_cellprofiler(
         analysis_run (bool, optional): will use functions to complete an analysis pipeline (default is False)
 
     """
-    print(f"Starting CellProfiler run on {pathlib.Path(path_to_images).name}")
-    # A log file is created for each plate or data set name (based on folder name with images)
-    with open(pathlib.Path(f"logs/cellprofiler_output_{pathlib.Path(path_to_images).name}.log"), "w") as cellprofiler_output_file:
-        # run CellProfiler for a illumination correction pipeline
-        command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
-        subprocess.run(command, shell=True, stdout=cellprofiler_output_file, stderr=cellprofiler_output_file)
-    cellprofiler_output_file.close()
+    # run CellProfiler illumination correction pipeline
+    if not analysis_run:
+        print(f"Starting CellProfiler run on {pathlib.Path(path_to_images).name}")
+        # A log file is created for each plate or data set name (based on folder name with images)
+        with open(pathlib.Path(f"logs/cellprofiler_output_{pathlib.Path(path_to_images).name}.log"), "w") as cellprofiler_output_file:
+            # run CellProfiler for a illumination correction pipeline
+            command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
+            subprocess.run(command, shell=True, stdout=cellprofiler_output_file, stderr=cellprofiler_output_file)
```

Comment:
Consider using `subprocess`'s list-based input instead of a string from the command. Doing this would enable you to use the variables found within brackets in the f-string directly instead of needing to wrap them. It also might mean the formatting of the command becomes more easily reviewed (the list elements may take multiple lines).

Reply:
```suggestion
            command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}".split()
            subprocess.run(command, stdout=cellprofiler_output_file, stderr=cellprofiler_output_file)
```
Is normally how I do it. By default `split()` will separate the strings via white spaces. Also, since we're passing a list of commands/parameters, this will require `shell=False` or just remove it since shell is set to `False` by [default](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module)

Reply:
I am going to go with Erik's suggestion but this will be flexible in case something else (like Dave's suggestion) works better.